### PR TITLE
LD-1: Base Repo

### PR DESCRIPTION
**Pull request by Foltz124**
Setup the general directory structure for the project, created some basic player and die classes, and then set up check unit testing files.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 2.8.12)
+
+project(LiarsDice)
+
+# Root Parameters
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_BUILD_TYPE Debug)
+set(BUILD_TESTS ON)
```
Comment (Foltz124) [inline CMakeLists.txt:8]: Default to OFF

File: lib/core/player/Player.c [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "Player.h"
+
+Player* buildPlayer(char* name, unsigned int tokens) {
+    Player* player = malloc(sizeof(Player));
+    if(player) {
```
Comment (lastranget) [inline lib/core/player/Player.c:8]: General c question. Why do you need to check if player?

Comment (Foltz124) [reply lib/core/player/Player.c:8]: If we fail to reserve memory, malloc will return NULL. So we need to check that we aren't working with a null pointer. That check if equivalent to:
```c
if(player != NULL) {
    ....
}
```

**Comment by lastranget** [comment]
This all looks good. What a great start

**Comment by Foltz124** [comment]
[Here](https://en.m.wikipedia.org/wiki/Opaque_pointer) is something I found that we might want to use. It's effectively a way to get encapsulation in C. I'm not sure if it's overkill or not, but might want to update these with that in mind. It would probably simplify the linked list a bit.

**Comment by Foltz124** [comment]
> [Here](https://en.m.wikipedia.org/wiki/Opaque_pointer) is something I found that we might want to use. It's effectively a way to get encapsulation in C. I'm not sure if it's overkill or not, but might want to update these with that in mind. It would probably simplify the linked list a bit. 

This example doesn't use the way I've seen, but typically you just typedef a void pointer, and that allows you to hide strict details in the implementation file

**Pull request merged**